### ISSUE TITLE
ToDo: diffs FF56-FF57

### ISSUE THREAD

**Issue by earthlng** [ghacksuserjs/ghacks-user.js#254]
[**v56.0 vs v57.0**](https://github.com/earthlng/FFprefs-diffs/blob/master/diffs/5x/diff-v56.0-vs-v57.0.log.js) - 161 diffs ( 80 new, 33 gone, 48 different )

FF57 is scheduled for release 2017-11-14

## new in v57.0:

* [x] pref("browser.library.activity-stream.enabled", true); [1354536](https://bugzilla.mozilla.org/show_bug.cgi?id=1354536) - https://github.com/ghacksuserjs/ghacks-user.js/commit/06d9d61558543b9a0ce80bd45f431a81574f16bc
* [x] pref("browser.ping-centre.telemetry", true); [1390249](https://bugzilla.mozilla.org/show_bug.cgi?id=1390249) - https://github.com/ghacksuserjs/ghacks-user.js/commit/22e51e2a083db23e19cf477cef7dc0877240cc6b
* [x] pref("browser.urlbar.maxHistoricalSearchSuggestions", 0); [1181644](https://bugzilla.mozilla.org/show_bug.cgi?id=1181644) - https://github.com/ghacksuserjs/ghacks-user.js/commit/64a6eb794aee53277acef15af22898c44f78b606
* [x] telemetry pings - https://github.com/ghacksuserjs/ghacks-user.js/commit/1dd683c81f9512bfebabb4037dd1b58bff700fb2
   * pref("toolkit.telemetry.bhrPing.enabled", true); [1380081](https://bugzilla.mozilla.org/show_bug.cgi?id=1380081)
   * pref("toolkit.telemetry.firstShutdownPing.enabled", true); [1390095](https://bugzilla.mozilla.org/show_bug.cgi?id=1390095)
* personal section
   * [x] pref("browser.download.autohideButton", true); [1397447](https://bugzilla.mozilla.org/show_bug.cgi?id=1397447) - https://github.com/ghacksuserjs/ghacks-user.js/commit/b373fe2a58747ffe9c48b2e6f3ede473d7a96469
   * [x] pref("browser.tabs.loadBookmarksInTabs", false); [658245](https://bugzilla.mozilla.org/show_bug.cgi?id=658245) - https://github.com/ghacksuserjs/ghacks-user.js/commit/76bc2bce252fc45e95aa0a58c0b5c8b8d0efa227
   * [x] pref("media.wmf.amd.vp9.enabled", true); - https://github.com/ghacksuserjs/ghacks-user.js/commit/3028a7a61a5819b38ad7143fffacb7510b134292
* privacy.resistFingerprinting.block_mozAddonManager (hidden pref) `4503` - [1384330](https://bugzilla.mozilla.org/show_bug.cgi?id=1384330)
* *leaving here for visibility*:
   * pref("browser.bookmarks.openInTabClosesMenu", true); [260611](https://bugzilla.mozilla.org/show_bug.cgi?id=260611)
=> this pref is wildly inconsistent and not ready for prime time

## removed, renamed or hidden in v57.0:

ALL DONE - https://github.com/ghacksuserjs/ghacks-user.js/commit/f89bb3f1c224e4a02523193c2cb7407415e6bcf5

* [x] pref("browser.bookmarks.showRecentlyBookmarked", true); [1401238](https://bugzilla.mozilla.org/show_bug.cgi?id=1401238)
* [x] pref("browser.casting.enabled", false); [1393582](https://bugzilla.mozilla.org/show_bug.cgi?id=1393582)
* [x] pref("devtools.webide.autoinstallFxdtAdapters", true); [1393497](https://bugzilla.mozilla.org/show_bug.cgi?id=1393497)
* [x] pref("media.eme.chromium-api.enabled", true); [1395468](https://bugzilla.mozilla.org/show_bug.cgi?id=1395468)
* [x] social prefs
   * [1388902](https://bugzilla.mozilla.org/show_bug.cgi?id=1388902) social.share.activationPanelEnabled & social.shareDirectory
   * [1406193](https://bugzilla.mozilla.org/show_bug.cgi?id=1406193) social.enabled, social.whitelist, social.directories, social.remote-install.enabled, social.toast-notifications.enabled - note: these are just missed prefs not cleaned out until 58

## changed in v57.0:

* [x] StorageManager API - https://github.com/ghacksuserjs/ghacks-user.js/commit/59ae0b1028d0fce03194c83a76db419fa5c3d0ac
   * pref("browser.storageManager.enabled", true); // prev: false `2706` [1399398](https://bugzilla.mozilla.org/show_bug.cgi?id=1399398)
   * pref("dom.storageManager.enabled", true); // prev: false `2706`
* pref("browser.newtabpage.activity-stream.enabled", true); // prev: false `0514`
* pref("browser.urlbar.speculativeConnect.enabled", true); // prev: false `0810` 
* pref("plugin.state.flash", 1); // prev: 2 `1803`
* pref("privacy.trackingprotection.ui.enabled", true); // prev: false `0421`
* *leaving here for visibility*:
  * pref("extensions.legacy.enabled", false); // prev: true
  * pref("dom.enable_performance_observer", true); // prev: false `should be covered by RFP`
  * on Linux `security.sandbox.content.level` moves from 2 to 3 as default (thanks @2glops )
  => see https://github.com/ghacksuserjs/ghacks-user.js/issues/254#issuecomment-344079045

### redundant in 57 due to RFP <sup>privacy.resistFingerprinting</sup>

* [x] these get moved to 4600 and renumbered after stable land - https://github.com/ghacksuserjs/ghacks-user.js/commit/89e0aed652e5ef13bd0c1ed4f78c4afb0cb79dee
  * 2506: spoof video statistics
  * 2509: disable touch events
* Presentation API
  * [1382533](https://bugzilla.mozilla.org/show_bug.cgi?id=1382533) added some FP resistance **but we keep it as `2513` for now**
* WebGL
   * [1217290](https://bugzilla.mozilla.org/show_bug.cgi?id=1217290) added some FP resistance to WebGL, ***but we choose to keep WebGL totally disabled***. Note: that currently WebGL can fail to initialize anyway - see [1409677](https://bugzilla.mozilla.org/show_bug.cgi?id=1409677)

---

## ignore

==NEW
```js
pref("accessibility.indicator.enabled", false);
pref("accessibility.support.url", "https://support.mozilla.org/%LOCALE%/kb/accessibility-services");
pref("apz.drag.touch.enabled", true);
pref("apz.keyboard.passive-listeners", true);
pref("browser.newtabpage.activity-stream.aboutHome.enabled", true);
pref("browser.newtabpage.activity-stream.prerender", true);
pref("browser.onboarding.notification.max-life-time-all-tours-ms", 1209600000);
pref("browser.onboarding.state", "default");
pref("browser.ping-centre.log", false);
pref("browser.ping-centre.production.endpoint", "https://tiles.services.mozilla.com/v3/links/ping-centre");
pref("browser.ping-centre.staging.endpoint", "https://onyx_tiles.stage.mozaws.net/v3/links/ping-centre");
pref("browser.safebrowsing.passwords.enabled", false);
pref("browser.tabs.extraDragSpace", false);
pref("browser.tabs.remote.warmup.enabled", false);
pref("browser.tabs.remote.warmup.maxTabs", 3);
pref("browser.tabs.remote.warmup.unloadDelayMs", 2000);
pref("browser.tabs.tabMinWidth", 76);
pref("devtools.debugger.features.async-stepping", true);
pref("devtools.debugger.features.wasm", true);
pref("devtools.debugger.project-text-search-enabled", true);
pref("devtools.enabled", true);
pref("devtools.netmonitor.persistlog", false);
pref("dom.abortController.enabled", true);
pref("dom.abortController.fetch.enabled", true);
pref("dom.event.coalesce_mouse_move", false);
pref("dom.ipc.scheduler", false);
pref("dom.ipc.scheduler.chaoticScheduling", false);
pref("dom.ipc.scheduler.preemption", false);
pref("dom.ipc.scheduler.threadCount", 2);
pref("dom.ipc.scheduler.useMultipleQueues", true);
pref("dom.ipc.useNativeEventProcessing.content", true);
pref("dom.max_ext_content_script_run_time", 5);
pref("dom.payments.loglevel", "Warn");
pref("dom.serviceWorkers.disable_open_click_delay", 1000);
pref("dom.streams.enabled", false);
pref("extensions.formautofill.creditCards.available", false);
pref("gfx.webrender.blob-images", false);
pref("gfx.webrender.debug.profiler", false);
pref("gfx.webrender.debug.render-targets", false);
pref("gfx.webrender.debug.texture-cache", false);
pref("identity.fxaccounts.remote.email.uri", "https://accounts.firefox.com/?service=sync&context=fx_desktop_v3&action=email");
pref("image.cache.factor2.threshold-surfaces", 4);
pref("input_event_queue.count_for_prediction", 9);
pref("input_event_queue.default_duration_per_event", 1);
pref("input_event_queue.duration.max", 8);
pref("input_event_queue.duration.min", 1);
pref("input_event_queue.supported", true);
pref("javascript.options.mem.gc_allocation_threshold_factor", 90);
pref("javascript.options.mem.gc_allocation_threshold_factor_avoid_interrupt", 90);
pref("javascript.options.wasm_ionjit", true);
pref("jsloader.shareGlobal", true);
pref("layers.mlgpu.enabled", true);
pref("layers.omtp.enabled", false);
pref("layers.omtp.force-sync", false);
pref("layout.css.ruby.intercharacter.enabled", false);
pref("layout.css.stylo-blocklist.blocked_domains", "");
pref("layout.css.stylo-blocklist.enabled", false);
pref("media.navigator.streams.fake", false); // WebRTC
pref("media.wmf.deblacklisting-for-telemetry-in-gpu-process", true);
pref("network.cookie.ipc.sync", false);
pref("network.http.rcwn.min_wait_before_racing_ms", 0);
pref("network.http.tailing.delay-max", 6000);
pref("network.http.tailing.delay-quantum", 600);
pref("network.http.tailing.delay-quantum-after-domcontentloaded", 100);
pref("network.http.tailing.enabled", true);
pref("privacy.userContext.extension", "");
pref("security.use_sqldb", false);
pref("services.sync.prefs.sync.browser.safebrowsing.downloads.enabled", true);
pref("services.sync.prefs.sync.browser.safebrowsing.passwords.enabled", true);
pref("urlclassifier.passwordAllowTable", "goog-passwordwhite-proto");
pref("webgl.1.allow-core-profiles", false);
```

==REMOVED or HIDDEN
```js
pref("accessibility.loadedInLastSession", false);
pref("app.support.e10sAccessibilityUrl", "https://support.mozilla.org/1/firefox/%VERSION%/%OS%/%LOCALE%/accessibility-ppt");
pref("browser.cache.auto_delete_cache_version", 0);
pref("browser.cache.use_new_backend", 0);
pref("browser.cache.use_new_backend_temp", true);
pref("browser.customizemode.tip0.learnMoreUrl", "https://support.mozilla.org/1/firefox/%VERSION%/%OS%/%LOCALE%/customize");
pref("browser.customizemode.tip0.shown", false);
pref("browser.onboarding.hidden", false);
pref("browser.photon.structure.enabled", false);
pref("browser.preferences.useOldOrganization", false);
pref("devtools.debugger.chrome-debugging-port", 6080);
pref("devtools.loader.hotreload", false);
pref("devtools.styleeditor.source-maps-enabled", true);
pref("devtools.webide.adaptersAddonID", "fxdevtools-adapters@mozilla.org");
pref("devtools.webide.adaptersAddonURL", "https://ftp.mozilla.org/pub/mozilla.org/labs/valence/#OS#/valence-#OS#-latest.xpi");
pref("devtools.webide.addonsURL", "https://ftp.mozilla.org/pub/mozilla.org/labs/fxos-simulator/index.json");
pref("devtools.webide.logSimulatorOutput", false);
pref("devtools.webide.simulatorAddonID", "fxos_#SLASHED_VERSION#simulator@mozilla.org");
pref("devtools.webide.simulatorAddonRegExp", "fxos(.*)_simulator@mozilla\.org$");
pref("devtools.webide.simulatorAddonsURL", "https://ftp.mozilla.org/pub/mozilla.org/labs/fxos-simulator/#VERSION#/#OS#/fxos_#SLASHED_VERSION#_simulator-#OS#-latest.xpi");
pref("gfx.webrender.profiler.enabled", false);
pref("layers.mlgpu.dev-enabled", false);
pref("layout.css.clip-path-shapes.enabled", true);
pref("media.wakelock_timeout", 2000);
pref("network.http.keep_empty_response_headers_as_empty_string", true);
pref("security.webauth.u2f_enable_softtoken", false);
pref("security.webauth.u2f_enable_usbtoken", false);
```

==CHANGED
```js
pref("apz.autoscroll.enabled", true); // prev: false
pref("apz.frame_delay.enabled", true); // prev: false
pref("apz.keyboard.enabled", true); // prev: false
pref("browser.onboarding.newtour", "performance,private,screenshots,addons,customize,default"); // prev: "private,addons,customize,search,default,sync"
pref("browser.onboarding.tourset-version", 2); // prev: 1
pref("browser.onboarding.updatetour", "performance,library,screenshots,singlesearch,customize,sync"); // prev: ""
pref("browser.preferences.offlineGroup.enabled", false); // prev: true
  // ^^ storage management - 1399398
pref("browser.safebrowsing.provider.google.advisoryName", "Google Safe Browsing"); // prev: "Google Safe Browsing."
pref("browser.safebrowsing.provider.google4.advisoryName", "Google Safe Browsing"); // prev: "Google Safe Browsing."
pref("browser.safebrowsing.provider.google4.lists", "<long-string>"); // prev: "<long-string>"
pref("browser.search.widget.inNavBar", false); // prev: true
pref("browser.slowStartup.timeThreshold", 20000); // prev: 30000
pref("browser.touchmode.auto", true); // prev: false
pref("devtools.hud.loglimit", 10000); // prev: 1000
pref("devtools.webconsole.new-frontend-enabled", true); // prev: false
pref("dom.forms.datetime", true); // prev: false
pref("font.name-list.monospace.ja", "MS Gothic, MS Mincho, Meiryo, Yu Gothic, Yu Mincho, MS PGothic, MS PMincho"); // prev: "MS Gothic, MS Mincho, MS PGothic, MS PMincho,Meiryo"
pref("font.name-list.sans-serif.ja", "Meiryo, Yu Gothic, MS PGothic, MS Gothic, Yu Mincho, MS PMincho, MS Mincho"); // prev: "MS PGothic, MS Gothic, MS PMincho, MS Mincho,Meiryo"
pref("font.name-list.sans-serif.ko", "Gulim, Malgun Gothic"); // prev: "Gulim"
pref("font.name-list.serif.ja", "Yu Mincho, MS PMincho, MS Mincho, Meiryo, Yu Gothic, MS PGothic, MS Gothic"); // prev: "MS PMincho, MS Mincho, MS PGothic, MS Gothic,Meiryo"
pref("javascript.options.mem.nursery.max_kb", 16384); // prev: -1
pref("layers.mlgpu.enable-on-windows7", true); // prev: false
pref("layout.css.servo.enabled", true); // prev: false
pref("lightweightThemes.recommendedThemes", "<long-string>"); // prev: "<long-string>"
pref("media.suspend-bkgnd-video.enabled", true); // prev: false
pref("media.windows-media-foundation.use-nv12-format", true); // prev: false
pref("media.wmf.disable-d3d11-for-dlls", "<long-string>"); // prev: "<long-string>"
pref("memory.free_dirty_pages", true); // prev: false
pref("network.IDN.restriction_profile", "high"); // prev: "moderate"
pref("network.preload", false); // prev: true
pref("network.standard-url.punycode-host", true); // prev: false
pref("preferences.allow.omt-write", true); // prev: false
pref("privacy.popups.disable_from_plugins", 3); // prev: 2
pref("security.data_uri.unique_opaque_origin", true); // prev: false
pref("security.webauth.webauthn_enable_usbtoken", true); // prev: false
pref("urlclassifier.disallow_completions", "<long-string>"); // prev: "<long-string>"
pref("urlclassifier.downloadAllowTable", "goog-downloadwhite-proto"); // prev: "goog-downloadwhite-digest256"
pref("urlclassifier.downloadBlockTable", "goog-badbinurl-proto"); // prev: "goog-badbinurl-shavar"
pref("urlclassifier.malwareTable", "<long-string>"); // prev: "<long-string>"
pref("urlclassifier.phishTable", "goog-phish-proto,test-phish-simple"); // prev: "goog-phish-shavar,test-phish-simple"
```

**Comment by earthlng** [comment]
@Thorin-Oakenpants I wrote a script that finds the tickets for most pref changes. Once we've moved more stuff to ignore I'll paste a list with all the bugzilla links.

**Comment by Thorin-Oakenpants** [comment]
^^^ excellent work number 2 ... I am ignoring this thread until I sort out
- the extensions+cookies issue (wiki, readme etc) re storage/IDB
- the new/revamped persistent storage section issue - which will also close the strategy issue
- the cache+ETAGs issue

Until then, have fun in here playing on your own :) Am off to watch [Spiderpig](http://www.imdb.com/title/tt2250912/)

```
Spider Pig!
Spider Pig!
Does whatever a spider pig does!
Can he swing from a web?

No he can't
He's a pig
Lookout!
He is the spider pig!!
```

**Comment by Atavic** [comment]
[https://support.mozilla.org/kb/accessibility-services](Accessibility services are primarily consumed by accessibility clients such as screen reading applications designed to help users with vision impairments).

```
pref("browser.ping-centre.production.endpoint", "https://127.0.0.1");
pref("browser.ping-centre.staging.endpoint", "https://127.0.0.1");
```

Safely removing the URL? Even change it to `http://` ?

**Comment by Atavic** [comment]
[https://support.mozilla.org/kb/accessibility-services](Accessibility services are primarily consumed by accessibility clients such as screen reading applications designed to help users with vision impairments).

`
pref("browser.ping-centre.production.endpoint", "https://127.0.0.1");
pref("browser.ping-centre.staging.endpoint", "https://127.0.0.1");
`

Safely removing the URL? Even change it to `http://` ?

**Comment by Atavic** [comment]
[https://support.mozilla.org/kb/accessibility-services](Accessibility services are primarily consumed by accessibility clients such as screen reading applications designed to help users with vision impairments).

```
pref("browser.ping-centre.production.endpoint", "https://127.0.0.1");
pref("browser.ping-centre.staging.endpoint", "https://127.0.0.1");
```

...safely removing the URL?

**browser.onboarding** is a multifaceted mess with a *notification.max-time* in unix epoch format: `1` should work fine there.

**dom.ipc.scheduler** is related to [this](https://wiki.mozilla.org/Quantum/DOM) and handles tabs like chrome, deferring some heavy loading and prioritising the lighter ones.

**Comment by Atavic** [comment]
[Accessibility services are primarily consumed by accessibility clients such as screen reading applications designed to help users with vision impairments](https://support.mozilla.org/kb/accessibility-services).

```
pref("browser.ping-centre.production.endpoint", "https://127.0.0.1");
pref("browser.ping-centre.staging.endpoint", "https://127.0.0.1");
```

...safely removing those URLs?

**dom.ipc.scheduler** is related to [this](https://wiki.mozilla.org/Quantum/DOM) and handles tabs like chrome, deferring some heavy loading and prioritising the lighter ones.

**Comment by Atavic** [comment]
@earthling Side by side mode [here](https://cat-in-136.github.io/2017/10/diff-between-firefox-570-beta-3-default.html).

**Comment by Thorin-Oakenpants** [comment]
Only 12 days to go boys and girls ... r u excited?

**Comment by Thorin-Oakenpants** [comment]
changed
- `security.mixed_content.send_hsts_priming` false=>true. See 1242. Since 51, is this now ready for us to ALSO force as true. Our reasoning behind it being false was delay issues. Zilla's metrics should be good to go, right?

**Comment by earthlng** [comment]
> six FF releases later, are we ready to turn this to true like Zilla are

it's false again in beta12. regardless of what zilla does, I would personally never use this anyway because I don't agree with the idea that the mixed-content-blocker should send out priming requests. delay or not doesn't matter to me.

**Comment by earthlng** [comment]
I know the song but only after I looked it up on youtube, didn't know the name of the band though.

**Comment by Thorin-Oakenpants** [comment]
FYI: https://medium.com/georg-fritzsche/data-preference-changes-in-firefox-58-2d5df9c428b5

**Comment by earthlng** [comment]
57b13 changes since 57b3

#### new

pref("accessibility.indicator.enabled", false); // new with value 'true' in Beta 3
pref("browser.tabs.tabMinWidth", 76);
pref("browser.urlbar.maxHistoricalSearchSuggestions", 0); // new with value 1 in Beta 3
pref("dom.serviceWorkers.disable_open_click_delay", 1000);
pref("identity.fxaccounts.remote.email.uri", "https://accounts.firefox.com/?service=sync&context=fx_desktop_v3&action=email");
pref("layout.css.stylo-blocklist.blocked_domains", "");
pref("layout.css.stylo-blocklist.enabled", false);
~~pref("services.sync.engine.bookmarks.validation.enabled", true);~~

#### removed, renamed or hidden

pref("devtools.webide.addonsURL", "https://ftp.mozilla.org/pub/mozilla.org/labs/fxos-simulator/index.json");
pref("devtools.webide.logSimulatorOutput", false);
pref("devtools.webide.simulatorAddonID", "fxos_#SLASHED_VERSION#_simulator@mozilla.org");
pref("devtools.webide.simulatorAddonRegExp", "fxos_(.*)_simulator@mozilla\\.org$");
pref("devtools.webide.simulatorAddonsURL", "https://ftp.mozilla.org/pub/mozilla.org/labs/fxos-simulator/#VERSION#/#OS#/fxos_#SLASHED_VERSION#_simulator-#OS#-latest.xpi");
~~pref("geo.provider.ms-windows-location", false);~~

#### changed

~~pref("browser.crashReports.unsubmittedCheck.enabled", true); // prev: false~~
~~pref("editor.use_div_for_default_newlines", true); // prev: false~~
~~pref("geo.wifi.uri", "https://location.services.mozilla.com/v1/geolocate?key=%MOZILLA_API_KEY%"); // prev: "https://www.googleapis.com/geolocation/v1/geolocate?key=%GOOGLE_API_KEY%"~~
pref("lightweightThemes.recommendedThemes", "<long-string>"); // prev: "<long-string>"
pref("media.suspend-bkgnd-video.enabled", true); // prev: false
pref("network.IDN.restriction_profile", "high"); // prev: "moderate"
pref("network.preload", false); // prev: true
~~pref("security.data_uri.block_toplevel_data_uri_navigations", true); // prev: false~~
~~pref("security.mixed_content.send_hsts_priming", true); // prev: false~~
~~pref("security.mixed_content.use_hsts", true); // prev: false~~

**Comment by earthlng** [comment]
* layout.css.stylo-blocklist.blocked_domains
Bug [1403077](https://bugzilla.mozilla.org/show_bug.cgi?id=1403077) - Part 1: Implement the stylo blocklist mechanism. r=heycam, a=ritu

* layout.css.stylo-blocklist.enabled
Bug [1411517](https://bugzilla.mozilla.org/show_bug.cgi?id=1411517) - stylo: &quot;layout.css.stylo-blocklist.enabled&quot; pref's default value should be false. r=xidorn, a=ritu
Bug [1403077](https://bugzilla.mozilla.org/show_bug.cgi?id=1403077) - Part 1: Implement the stylo blocklist mechanism. r=heycam, a=ritu

* network.IDN.restriction_profile
Bug [1399939](https://bugzilla.mozilla.org/show_bug.cgi?id=1399939) - Switch to highly restrictive profile for IDN. r=dveditz, r=jfkthame, a=sledru

**Comment by Thorin-Oakenpants** [comment]
redundant in 57 due to RFP
- WebGL: my thoughts on this are that WebGL stays blocked (2010-2012 various prefs). After 57 lands, and I/we find time maybe we could revisit this. Thoughts?

**Comment by earthlng** [comment]
I agree that WebGL should remain disabled. The Presentation API is weird as well IMO, too many prefs to move them all to the RFP alternatives section. I would keep that where it currently is.

**Comment by Thorin-Oakenpants** [comment]
<details><summary>moved from new to ignore</summary><p>

```js
pref("accessibility.indicator.enabled", false);
pref("apz.drag.touch.enabled", true);
pref("apz.keyboard.passive-listeners", true);
pref("browser.bookmarks.openInTabClosesMenu", true);
pref("browser.onboarding.notification.max-life-time-all-tours-ms", 1209600000);
pref("browser.onboarding.state", "default");
pref("browser.tabs.extraDragSpace", false);
pref("browser.tabs.loadBookmarksInTabs", false);
pref("browser.tabs.remote.warmup.enabled", false);
pref("browser.tabs.remote.warmup.maxTabs", 3);
pref("browser.tabs.remote.warmup.unloadDelayMs", 2000);
pref("browser.tabs.tabMinWidth", 76);
pref("devtools.debugger.features.async-stepping", true);
pref("devtools.debugger.features.wasm", true);
pref("devtools.debugger.project-text-search-enabled", true);
pref("dom.abortController.enabled", true);
pref("dom.abortController.fetch.enabled", true);
pref("dom.event.coalesce_mouse_move", false);
pref("dom.ipc.scheduler", false);
pref("dom.ipc.scheduler.chaoticScheduling", false);
pref("dom.ipc.scheduler.preemption", false);
pref("dom.ipc.scheduler.threadCount", 2);
pref("dom.ipc.scheduler.useMultipleQueues", true);
pref("dom.ipc.useNativeEventProcessing.content", true);
pref("dom.max_ext_content_script_run_time", 5);
pref("gfx.webrender.blob-images", false);
pref("gfx.webrender.debug.profiler", false);
pref("gfx.webrender.debug.render-targets", false);
pref("gfx.webrender.debug.texture-cache", false);
pref("image.cache.factor2.threshold-surfaces", 4);
pref("input_event_queue.count_for_prediction", 9);
pref("input_event_queue.default_duration_per_event", 1);
pref("input_event_queue.duration.max", 8);
pref("input_event_queue.duration.min", 1);
pref("input_event_queue.supported", true);
pref("javascript.options.mem.gc_allocation_threshold_factor", 90);
pref("javascript.options.mem.gc_allocation_threshold_factor_avoid_interrupt", 90);
pref("layers.mlgpu.enabled", true);
pref("layers.omtp.enabled", false);
pref("layers.omtp.force-sync", false);
pref("layout.css.ruby.intercharacter.enabled", false);
pref("layout.css.stylo-blocklist.blocked_domains", "");
pref("layout.css.stylo-blocklist.enabled", false);
pref("media.navigator.streams.fake", false); // WebRTC
pref("media.wmf.deblacklisting-for-telemetry-in-gpu-process", true);
pref("network.cookie.ipc.sync", false);
pref("network.http.rcwn.min_wait_before_racing_ms", 0);
pref("network.http.tailing.delay-max", 6000);
pref("network.http.tailing.delay-quantum", 600);
pref("network.http.tailing.delay-quantum-after-domcontentloaded", 100);
pref("network.http.tailing.enabled", true);
pref("webgl.1.allow-core-profiles", false);
```

</p></details>

**Comment by Thorin-Oakenpants** [comment]
@fmarier `browser.safebrowsing.passwords.enabled` [1385484](https://bugzilla.mozilla.org/show_bug.cgi?id=1385484) - I cannot understand a word of this. Can you ELI5 please?

**Comment by earthlng** [comment]
Wow you moved lots of stuff to ignore, nice, thx.

* `network.IDN.restriction_profile` - this increases the times when punycode is shown but we already show punycode all the time so we don't need this
* the 2 `devtools.*` under new: we shouldn't mess with the .enabled one ATM IMO and the .persistlog was just split out from devtools.webconsole.persistlog to allow to change them individually.
* `privacy.userContext.extension` - stores the ID (? - I assume) from a container addon if one is installed - no need to mess with this, best to ignore IMO.
* `pref("browser.bookmarks.openInTabClosesMenu", true);` + `pref("browser.tabs.loadBookmarksInTabs", false);` - you moved them to ignore but I think the 1st one would be nice to have in the Personal section (it's currently very annoying IMO that the menu closes when I open a bookmark) and the 2nd I'm not sure what it does exactly but is worth checking
* the ping-centre stuff - it looks like this is only used by activity-stream which we already disable. And it's also never sending anything if `datareporting.healthreport.uploadEnabled = false`. We could add the .telemetry one as `0514b` but ignoring it is fine as well IMO.
* `identity.fxaccounts.remote.email.uri` - is part of sync which we generally ignore
* the 3 activity-stream crap with checkboxes under new: we probably don't need the first 2 but maybe the "library" one. We'll have to see once it's released.

**Comment by Thorin-Oakenpants** [comment]
`network.IDN.restriction_profile` - I disagree. Not everyone will want to have `network.IDN_show_punycode` => false due to their language, so this is a fallback. For 57+ users the default is now `high`, but enforcing it can;t hurt - does this change also take place in ESR?

- https://reviewboard.mozilla.org/r/184586/diff/2/#index_header (the link in that no longer exists)
- https://www.unicode.org/reports/tr36/index.html#Security_Levels_and_Alerts

**Comment by Thorin-Oakenpants** [comment]
Thanks @fmarier .. I'll move it to ignore then :)  PS: what's the password!!! xD ... aaaaand, I still don't understand it .. maybe you can do a blog on it

**Comment by fmarier** [comment]
Yes, you can ignore it until that pref goes from `false` to `true` next year.

The "password" part is because it aims to defeat phishing pages that steal passwords (e.g. fake login pages).

I will definitely blog about it once we've landed all of the pieces :)

**Comment by Thorin-Oakenpants** [comment]
note: added `privacy.trackingprotection.ui.enabled` to deprecated

**Comment by Thorin-Oakenpants** [comment]
hey .. the `edited by X` has a dropdown history now

**Comment by Thorin-Oakenpants** [comment]
hey .. the `edited by X` has a dropdown history now

**Comment by earthlng** [comment]
^^ why?? it even still exists in the version of 58 I have

**Comment by Thorin-Oakenpants** [comment]
^ because it only landed uber recently - ask @fmarier

**Comment by fmarier** [comment]
The default for `privacy.trackingprotection.ui.enabled` changed to `true` in 57. We'll probably remove it at some point, but probably not before 59 or 60.

**Comment by earthlng** [comment]
yeah it now defaults to true and therefore the pref obviously still exists and does something. We can remove it from the user.js if that's what you want to do but it's certainly not deprecated.

**Comment by Thorin-Oakenpants** [comment]
my bad - https://reviewboard.mozilla.org/r/196362/diff/1/#index_header - not paying attention - that's not the prefs all js file !!

**Comment by earthlng** [comment]
np, that's why I prefer to look at the diffs on **hg**.mozilla.org (https://bugzilla.mozilla.org/show_bug.cgi?id=1393627#c17) Those also work without JS unlike the reviewboard.

**Comment by earthlng** [comment]
> The only pref we have in 0513 is for newtabpage - this does not look like a master switch to me. 

yes it looks weird but `browser.newtabpage.activity-stream.enabled` is definitely the master switch.

`browser.newtabpage.activity-stream.aboutHome.enabled`: the patch is at https://hg.mozilla.org/mozilla-central/rev/0ae745e100aa. This pref is only checked twice, the first time [here](https://hg.mozilla.org/mozilla-central/rev/0ae745e100aa#l2.68) *after* the main switch, and the 2nd time [here](https://hg.mozilla.org/mozilla-central/rev/0ae745e100aa#l2.101) where it is immediately followed by a check for the master switch. So in both cases the .aboutHome is kinda unnecessary with the main switch already disabled.

**Comment by Thorin-Oakenpants** [comment]
So I spose I better remove that pref then

**Comment by Thorin-Oakenpants** [comment]
I'll edit this as I keep whittling the last few items down
- pref("dom.payments.loglevel", "Warn"); => ignore => [1381186](https://bugzilla.mozilla.org/show_bug.cgi?id=1381186) => here's a gif of it in [action](http://g.recordit.co/VI8f1wv8AI.gif)

**Comment by Thorin-Oakenpants** [comment]
Major Pants to Earth Control....

```
/* 5008: open bookmarks in a new tab (FF57+)
 * [NOTE] you can also do this with a middle-click and from the context menu ***/
   // user_pref("browser.tabs.loadBookmarksInTabs", true)
/* 5009: leave the library bookmarks menu open after clicking a bookmark (FF57+) ***/
   // user_pref("browser.bookmarks.openInTabClosesMenu", false); 
```

Can you do some testing please in your shiny 58?

If 5008 (use-new-tab) is default false, and 5009 (menu) is true - where do the bookmarks open? I will assume they open exactly as they should by the method (ctrl-click, middle-click, left-click). I only ask because of [this](https://bugzilla.mozilla.org/show_bug.cgi?id=260611#c129) comment which says

> leave menu open for single bookmark menuitem clicks that ***open in new tab***

But that is not what the [code](https://hg.mozilla.org/mozilla-central/rev/53a1b44951cc#l1.12) says/implies

> Whether menu should close after ctrl-click, middle-click, etc.

more Q's
- do we need to explain the menu is the hamburger button>library>bookmarks ? i.e it is NOT the menu bar> bookmarks. I added the word library to the description if you think that's ok
- does this also work on library>recent history links? I'm going to guess yes in which case I should change the description
- how does it disappear - do you just click off?
- should either or both be active or inactive: both would be set to NON-default settings in the js

**Comment by earthlng** [comment]
Tested in FF nightly 58 (2017-09-21)

IDK if the behaviour is the same in 57 and/or in newer versions of 58 nightly !

the default values for both prefs are still the same as they are here for 57, openInTabClosesMenu = true and loadBookmarksInTabs = false

with only `browser.bookmarks.openInTabClosesMenu = false`:

* menubar > bookmarks > 
  * stays open when:
    * middle click
    * right click -> open in new tab
  * closes menu with ...
    * normal click
    * right click -> open
    * right click -> open in new (private) window

doesn't work at all for menubar > history

--- 

* via the toolbar icon >
  * menu stays open with
    * right click -> open !!
    * right click -> open in new tab
  * closes menu with ...
    * middle click !!
    * normal click
    * right click -> open in new (private) window

and it also works the same for the history items in that menu.
It also works exactly the same when accessing the "library" via the hamburger menu

- ctrl+click behaves exactly like middle click ie keeps the bookmarks menubar open but closes it in the "library"
- shift+click opens in a new window of the same type (normal or private) and therefore always closes the menu bar

`browser.tabs.loadBookmarksInTabs = true` basically changes "normal click" behaviour to "open in new tab" but it does **NOT** keep the menu open.

2 cents:

Since I mostly use the bookmarks in the menu bar + middle click + would like the menu to remain open I'll definitely use the 1st pref but don't really need the 2nd pref.
IMO it would be nice if the 2nd pref would keep the menu open and if middle click in the library would also keep the menu open.

> how does it disappear - do you just click off?

you just click somewhere outside of the menu or close it by clicking on bookmarks or the toolbar icon again.

> should either or both be active or inactive: both would be set to NON-default settings in the js

inactive for both is fine IMO.

**Comment by Thorin-Oakenpants** [comment]
```js
/* 5009: leave the bookmarks menu open after ~~clicking~~ right-clicking a bookmark (FF57+)
 * Applies to: ~~Menu Bar>Bookmarks~~, Bookmarks Toolbar icon, History Toolbar icon,
               Hamburger Menu>Library>Bookmarks, Hamburger Menu>Library>History
   // user_pref("browser.bookmarks.openInTabClosesMenu", false); 
```

So I just started typing this up and now I'm confused. From your tests a NORMAL click closes the menu - WTF? Maybe I should read the ticket  - is this only **meant** to stay open **only** if you use context menus (ignore the menubar)? If so that kinda makes sense (see mods in js in this post)

---
> browser.tabs.loadBookmarksInTabs = true changes "normal click" behaviour to "open in new tab" but it does NOT keep the menu open.

OK - since I've kinda worked out (I can't be arsed reading the ticket TBH) that 5009 above is all about right clicks (or rather context menus, cuz people can swap their mouse buttons) - then this (5008) has no effect on 5009 re sticky menus. For a minute I thought one interfered with the other.

Feel free to nit, fixup and propose a nice commit - or just do one

**Comment by earthlng** [comment]
To clarify, by "toolbar icon" I meant the one that's there by default, the one with the icon that looks like 4 books on a shelf. Under "customize" there are 2 items for "History" and "Bookmarks Menu" but I haven't tested those at all.

**Comment by Thorin-Oakenpants** [comment]
Oh .. can you test them? Also, I'll take a stab in the dark and say the "toolbar icon" you tested is called the Toolbar "Library icon" - I don't have one in 56 :)

**Comment by earthlng** [comment]
tooltip says "View history, saved bookmarks, and more" - I think we can safely call it "Library" :)

> Oh .. can you test them?

I sure can but the question is will I xD

Regardless, I think we shouldn't describe it in detail at all because it is likely gonna change again.
"bookmarks.openInTabClosesMenu" is kinda self-explanatory anyway if you ask me. Let users figure out the exact behavior for themselves if they decide to use it.

**Comment by earthlng** [comment]
`browser.urlbar.maxHistoricalSearchSuggestions` - I'm not sure what this does exactly but I think we don't need to worry about it because we have search suggestions disabled with 0807 + 0808.

**Comment by Thorin-Oakenpants** [comment]
> Regardless, I think we shouldn't describe it in detail at all 

Sheesh .. that's a 13yr old ticket - and long. It has a swag of issues still open - so it's not 100% polished anyway. I'll uber simplify it

**Comment by 2glops** [comment]
pref("browser.ping-centre.telemetry", true);
To be ignored IMO as toolkit.telemetry.enabled=false

pref("browser.urlbar.maxHistoricalSearchSuggestions", 0);
Ignore too.

**Comment by earthlng** [comment]
57.0 changes since 57b13

#### changed

~~pref("devtools.devedition.promo.enabled", true); // prev: false~~
~~pref("extensions.formautofill.available", "detect"); // prev: "staged-rollout"~~
pref("privacy.trackingprotection.ui.enabled", true); // prev: false
~~pref("security.tls.version.max", 4); // prev: 3~~

**Comment by 2glops** [comment]
Linux : 
"security.sandbox.content.level" moves from 2 to 3 as default

0 disables everything
1 filters dangerous system calls
2 additionally blocks writing to the filesystem
3 adds blocking of (most) reading from the filesystem.

http://www.morbo.org/2017/11/linux-sandboxing-improvements-in.html

No hurts here on Nightly, but could break some configurations. In that case, workaround is given in the above article.

**Comment by Thorin-Oakenpants** [comment]
ha! ... [1325923](https://bugzilla.mozilla.org/show_bug.cgi?id=1325923) - SVG loaded as `<img>` can use meta tags to set cookies ... [CVE](http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2017-7837)

but not if you block cookies by default (I assume) ... :)  You hearing me earthlng .. default deny bro

Fixed in 57 - both links currently denied/hidden

**Comment by Thorin-Oakenpants** [comment]
* [x] pref("browser.ping-centre.telemetry", true); -> ignore
* [x] browser.urlbar.maxHistoricalSearchSuggestions - see commit, added it for info
* [x] user_pref("gfx.layerscope.enabled", false); - **removed**, is default anyway, because it was left over when casting `2612` was deprecated. I also added this to the wiki script of removed items

I'll keep adding to this post ~~as~~ if I do more

**Comment by Thorin-Oakenpants** [comment]
OK guys .. we have 72 hours and then I am GONE .. I am away for five+ish days - get back on the 25th. And where I am going there is no internet and no cellphone coverage.. so 72 hrs to finish this and get the changelog done (although I am sure the changelog can be done without me) - would like this wrapped before I go (and I don't have much spare time)

**Comment by earthlng** [comment]
there are no prefs for the new permission UI for Camera, etc. In 57 it's just a frontend to edit the sqlite file

**Comment by Thorin-Oakenpants** [comment]
^^ So the settings button does what - tries to launch an SQL program? I don't follow. I assume then that it loads a window and populates it with data from places.sql where field type or something matches "_camera" or whatever? Makes no sense to me. Surely you mean it allows users to add/remove sites?

I know there are no prefs, but I might want to add the [SETTING] tag for the camera/mic so people can set excpetions.

**Comment by earthlng** [comment]
I think we should add `browser.ping-centre.telemetry` somewhere in the lower 0500's maybe 0506. There are traces of it on DXR for shield, onboarding and activity-stream and IMO we shouldn't rely that they won't use it for other stuff as well.

**Comment by Thorin-Oakenpants** [comment]
The other two pings relied on FHR - I thought you said this one did as well?

**Comment by earthlng** [comment]
It opens a window to change set permissions for those things, similar to the window that allows you to change cookie permissions.

**Comment by Thorin-Oakenpants** [comment]
^^ can I have a screenshot - a pic is worth 999 words

**Comment by earthlng** [comment]
> The other two pings relied on FHR - I thought you said this one did as well?

yes, atm but that might not always be the case especially when they decide to use it for other stuff as well.

**Comment by earthlng** [comment]
![unbenannt](https://user-images.githubusercontent.com/25846669/32877363-76edc1b6-caa1-11e7-8abb-c4524e2a07ff.png)

**Comment by Thorin-Oakenpants** [comment]
^^pic - ok, so that's location. Default is deny, websites ask, these are the exceptions

WHAT does camera and mic do - adding a site blocks? or allows? or either?

**Comment by earthlng** [comment]
![unbenannt](https://user-images.githubusercontent.com/25846669/32877406-a590b26c-caa1-11e7-8625-6f71ab2e8588.png)

**Comment by earthlng** [comment]
No, it's the same for all of them. FF58 will have new prefs to set the default:
```js
pref("permissions.default.camera", 0);
pref("permissions.default.desktop-notification", 0);
pref("permissions.default.geo", 0);
pref("permissions.default.microphone", 0);
pref("permissions.default.shortcuts", 0);
```

**Comment by Thorin-Oakenpants** [comment]
needs MOAR cowbell
```js
/* 0506: disable ?? what exactly??
 * [NOTE] currently blocked by X, this is covering your butt
 * [1] https://bugzilla.mozilla.org/show_bug.cgi?id=1390249 *** 
user_pref("browser.ping-centre.telemetry", false);
```

OK, we can do preferences UI changes and [SETTINGS] in 58 diffs since it will get picked up on there

**Comment by earthlng** [comment]
there's also another new pref option in the UI "Prevent accessibility services from accessing your browser" which we should probably add. It toggles `accessibility.force_disabled` to 1 or 0, 1=blocked.
The "Learn more" link opens https://support.mozilla.org/en-US/kb/accessibility-services

**Comment by Thorin-Oakenpants** [comment]
I've edited first post - go ahead and commit force_disabled and that ping - I'm off for a while, might come back in 6 or 12 hrs

Also - need to talk about WTF to do with the 2 or 3 storage prefs

**Comment by earthlng** [comment]
disable PingCentre telemetry (used in several system addons atm)
currently blocked by 'datareporting.healthreport.uploadEnabled'

**Comment by Thorin-Oakenpants** [comment]
^^ add `(see 0333)` and you're dandy

**Comment by Thorin-Oakenpants** [comment]
OMG: https://www.reddit.com/r/firefox/comments/7ddo3d/the_only_thing_i_dont_like_about_the_new_firefox/

**Comment by earthlng** [comment]
storageManager stuff: localStorage (or DOM Storage as you like to call it) is now cleared on shutdown with `privacy.clearOnShutdown.offlineApps` both in Release and ESR so IMO we should remove those 2 prefs from the user.js or at least make them inactive. Disabling them as we currently do also removes/hides the new (?) section in the Preferences UI (where the data could be controlled more granularly) which is not great. IMO we can/should also remove dom.storage.enabled (2705).
= 1 or 2 less FPing vectors by removing those and it can be cleared on shutdown or manually with either the built-in "Clear recent history" or an addon or the new UI option.

**Comment by Thorin-Oakenpants** [comment]
^^ THIS is a problem - it also wipes EXTENSION data - see #273 - we need to reverse the commit I just did

**Comment by Thorin-Oakenpants** [comment]
undid the commit - see you in 7 days

**Comment by earthlng** [comment]
```js
/* 2630: prevent accessibility services from accessing your browser
 * [SETTING] Options>Privacy & Security>Permissions>Prevent accessibility services from accessing your browser
 * [1] https://support.mozilla.org/kb/accessibility-services ***/
user_pref("accessibility.force_disabled", 1);
```

**Comment by earthlng** [comment]
@Thorin-Oakenpants what's the source for this part in `2706`:
> and even control whether or not they need to be alerted before the user agent disposes of site data in order to make room for other things.

??

**Comment by earthlng** [comment]
```js
pref("javascript.options.wasm_ionjit", true);
pref("jsloader.shareGlobal", true);
```
... not messing with this for now. Moved to ignore. When ESR59 is released we can check what TBB does with these 2.

**Comment by Thorin-Oakenpants** [comment]
Wait! so you want to allow 2422
```
user_pref("javascript.options.wasm", false);
```
but not include `javascript.options.wasm_ionjit` ?

> When ESR59 is released we can check what TBB does with these 2.

How will you remember to do this? DId you add them to the sticky items for follow up?

**Comment by earthlng** [comment]
> Wait! so you want to allow 2422

Where did I say that?
With javascript.options.wasm=false we don't need javascript.options.wasm_ionjit.

> How will you remember to do this?

brain 2.0

**Comment by Thorin-Oakenpants** [comment]
> Wait! so you want to allow 2422

Soz. Bad wording, I meant something else. As in we blocked wasm but do not futureproof for wasm_ionjit (yes I know one blocks the other). Also I doubt it will ever get flipped, so we'll never pick up on it in diffs - hence it should be added to the stick IMO. Unless you pinky swear you have a robust system to remember to check it in Tor when 59 rolls out, as you said. Brain 2.0 doesn't cut it, sorry.

At the very least we could just add it to 2422. yes/no?

Your call. Add to sticky or add to 2422?

**Comment by earthlng** [comment]
I've added it to the sticky

**Comment by Thorin-Oakenpants** [comment]
sweet :kiss:
